### PR TITLE
feat(nextjs): add support for custom next config and optimize style enhancer usages

### PR DESCRIPTION
**Pull request by jaysoo**
- Nextjs app schematic adds `next.config.js` by default with the config enhancer based on the chosen style (e.g. `withCSS`, `withSASS`, etc.) 
    - Users can modify and extend this config as needed.
- Add migration for existing apps to expose the `next.config.js` file that does what Nx used to do in the build builder (i.e. add support for *all* styles).

Related #2490

### CODE REVIEW

**Comment by jdpearce** [comment]
Holy carp. I wasn't expecting a migration as well. Nice 👍

File: packages/next/src/migrations/update-9-2-0/create-next-config.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return (host: Tree) => {
+    const workspaceJson = readWorkspace(host);
+    const nextProjects = Object.keys(workspaceJson.projects)
+      .map(name => {
+        const p = workspaceJson.projects[name];
+        const buildBuilder =
+          p.architect && p.architect.build && p.architect.build.builder;
+        return buildBuilder === '@nrwl/next:build' ? p : null;
+      })
+      .filter(Boolean);
```
Comment (jdpearce) [inline packages/next/src/migrations/update-9-2-0/create-next-config.ts]: How about optional chaining? 😁 

(might have to update prettier to get the formatting to play nice though)

```suggestion
    const nextProjects = Object.values<any>(workspaceJson.projects)
      .filter(p => p.architect?.build?.builder === '@nrwl/next:build');
```

You could, of course, do away with the `nextProjects` const entirely and just chain the above onto this `forEach`...

File: packages/next/src/utils/config.ts [hunk 2]
```diff
@@ -88,27 +89,24 @@ export function createWebpackConfig(
...
-  workspaceRoot: string,
-  projectRoot: string,
-  outputPath: string,
-  fileReplacements: FileReplacement[],
   phase:
     | typeof PHASE_PRODUCTION_BUILD
     | typeof PHASE_EXPORT
     | typeof PHASE_DEVELOPMENT_SERVER
-    | typeof PHASE_PRODUCTION_SERVER
+    | typeof PHASE_PRODUCTION_SERVER,
```
Comment (jdpearce) [inline packages/next/src/utils/config.ts]: Just a thought - why don't we turn this into an enum?

File: packages/next/src/migrations/update-9-2-0/create-next-config.spec.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    );
+  });
+
+  it('should create next.config.js if it does not exist', async () => {
+    tree.overwrite(
+      '/workspace.json',
+      JSON.stringify({
+        version: 1,
+        projects: {
+          demo: {
```
Comment (jdpearce) [inline packages/next/src/migrations/update-9-2-0/create-next-config.spec.ts]: Is it worth adding some dummy projects just to prove that the migration will actually skip over them? Maybe some projects without builders as well?

e.g.

```ts
          notnext: {
            root: 'apps/notnext',
            sourceRoot: 'apps/notnext/src',
            architect: {
              build: {
                builder: '@nrwl/web:build',
                options: {}
              }
            }
          },
          nobuilder: {
            root: 'apps/nobuilder',
            sourceRoot: 'apps/nobuilder/src',
            architect: {}
          }
```

**Review by catfireparty — COMMENTED** [review]
Looks good. Some minor comments, that's all. 👍

**Comment by jdpearce** [comment]
Hadn't looked at the failing tests, but of course the e2e tests will need updating... I didn't catch that 🤦‍♀️

File: packages/next/src/migrations/update-9-2-0/create-next-config.ts (hunk 1 above)
Comment (jaysoo) [reply packages/next/src/migrations/update-9-2-0/create-next-config.ts]: Yeah I thought about it.. maybe it's time to update prettier.

File: packages/next/src/utils/config.ts (hunk 2 above)
Comment (jaysoo) [reply packages/next/src/utils/config.ts]: Not sure if there's an advantage to that? The consts are coming from next itself.

**Comment by jaysoo** [comment]
> Hadn't looked at the failing tests, but of course the e2e tests will need updating... I didn't catch that 🤦‍♀️

Looks like somehow the `withCSS`, etc. isn't called correctly during e2e tests. I'll check it locally... it worked when I was running them with playground.

**Pull request merged**